### PR TITLE
feat: Native-image support

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: native-image-tests/src/main/scala/com/lightbend/EsbTester.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  sealed trait Command extends JsonSerializable
+
+  final case class Increase(amount: Int, replyTo: ActorRef[StatusReply[Increased]]) extends Command
+  @JsonCreator
+  final case class GetValue(replyTo: ActorRef[StatusReply[GetValueResponse]]) extends Command
+  final case class GetValueResponse(value: Int)
+
+  sealed trait Event extends JsonSerializable
+
+  // FIXME why doesn't @JsonCreator work as usual? is it something missing from the akka jackson feature?
```
Comment (patriknw) [inline native-image-tests/src/main/scala/com/lightbend/EsbTester.scala:29]: `-parameters` java compiler flag is needed for Java, but I thought that wasn't needed for Scala case classes. Compare that with the other Scala samples we have.

File: native-image-tests/src/main/resources/META-INF/native-image/ch.qos.logback/logback-classic/reflect-config.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    "methods":[{"name":"setParent","parameterTypes":["ch.qos.logback.core.spi.ContextAware"] }]
+  },
+  {
+    "name":"ch.qos.logback.core.pattern.PatternLayoutEncoderBase",
+    "methods":[{"name":"setPattern","parameterTypes":["java.lang.String"] }]
+  },
+  {
+    "name":"ch.qos.logback.core.spi.ContextAware",
+    "methods":[{"name":"valueOf","parameterTypes":["java.lang.String"] }]
+  }
```
Comment (patriknw) [inline native-image-tests/src/main/resources/META-INF/native-image/ch.qos.logback/logback-classic/reflect-config.json:63]: all this for logback is rather inconvenient, could we include it in some smart way but still not depend on logback?

**Review by patriknw — COMMENTED** [review]
looking good

File: native-image-tests/src/main/scala/com/lightbend/EsbTester.scala (hunk 1 above)
Comment (johanandren) [reply native-image-tests/src/main/scala/com/lightbend/EsbTester.scala:29]: Investigations on the Akka side shows that the parameter names are in the reflection metadata, but it still just doesn't work for single param constructors without the `@JsonProperty` :/ 

Fix/investigation PR: https://github.com/akka/akka/pull/32326

File: native-image-tests/src/main/resources/META-INF/native-image/ch.qos.logback/logback-classic/reflect-config.json (hunk 2 above)
Comment (johanandren) [reply native-image-tests/src/main/resources/META-INF/native-image/ch.qos.logback/logback-classic/reflect-config.json:63]: Yes, let's put it in akka-slf4j with reachability condition. I think users will still have to set some of the `--initialize-at-build-time=org.slf4j.LoggerFactory,ch.qos.logback` stuff in each project but that's less cumbersome.

File: native-image-tests/src/main/resources/application-h2-file.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+akka.persistence.journal.plugin = "akka.persistence.r2dbc.journal"
+akka.persistence.snapshot-store.plugin = "akka.persistence.r2dbc.snapshot"
+akka.persistence.state.plugin = "akka.persistence.r2dbc.state"
+
+akka.persistence.r2dbc.connection-factory = ${akka.persistence.r2dbc.h2}
+akka.persistence.r2dbc.connection-factory = {
+  protocol = "file"
+  database = "/tmp/test-h2-database"
+}
```
Comment (patriknw) [inline native-image-tests/src/main/resources/application-h2-file.conf:9]: trailing line break on config file (several of them)

File: docs/src/main/paradox/native-image.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Building Native Images
+
+Building native images with Akka HTTP is supported out of the box for the event sourced journal, snapshot store and
```
Comment (patriknw) [inline docs/src/main/paradox/native-image.md:3]: HTTP

**Review by patriknw — COMMENTED** [review]
looking good

File: native-image-tests/src/main/scala/com/lightbend/DurableStateTester.scala [hunk 5]
```diff
@@ -10,20 +10,18 @@ import akka.persistence.typed.PersistenceId
...
 
 import scala.concurrent.duration.DurationInt
 
 object DurableStateCounter {
   sealed trait Command extends JsonSerializable
   final case class Increase(amount: Int, replyTo: ActorRef[Increased]) extends Command
 
-  // FIXME why doesn't @JsonCreator work as usual? is it something missing from the jackson feature?
-  final case class GetState @JsonCreator() (replyTo: ActorRef[State]) extends Command
+  final case class GetState(replyTo: ActorRef[State]) extends Command
```
Comment (pvlugter) [inline native-image-tests/src/main/scala/com/lightbend/DurableStateTester.scala:20]: 👍🏼 great that it works out-of-the-box now

**Comment by johanandren** [comment]
Green test run in https://github.com/akka/akka-persistence-r2dbc/actions/runs/8068118973

**Review by pvlugter — APPROVED** [review]
LGTM

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**